tests/gdx-tests/src/com/badlogic/gdx/tests/scenes/scene2d/ui/DynamicChildActorsTest.java

```diff
@@ -0,0 +1,152 @@
+import com.badlogic.gdx.scenes.scene2d.ui.TextButton;
+import com.badlogic.gdx.scenes.scene2d.ui.Label.LabelStyle;
+import com.badlogic.gdx.scenes.scene2d.ui.TextButton.TextButtonStyle;
+import com.badlogic.gdx.scenes.scene2d.utils.Drawable;
+import com.badlogic.gdx.scenes.scene2d.utils.TextureRegionDrawable;
+import com.badlogic.gdx.tests.scenes.scene2d.ui.DynamicChildActorsTest.FlashingLabel.FlashingLabelStyle;
+import com.badlogic.gdx.tests.utils.GdxTest;
+
+/** Demonstrates the new g2d.ui interfaces.
+ * @author Daniels118 */
+public class DynamicChildActorsTest extends GdxTest {
+	private SpriteBatch batch;
+	private Stage stage;
+
+	@Override
+	public void create () {
+		Skin skin = new Skin(Gdx.files.internal("data/uiskin.json"));
+		batch = new SpriteBatch();
+		stage = new Stage();
+		Gdx.input.setInputProcessor(stage);
+		Table table = new Table(skin);
+		table.setFillParent(true);
+		stage.addActor(table);
+		
+		table.add("Default components").left().colspan(2);
+		table.row();
+		
+		Label label1 = new Label("Label 1", skin);
+		table.add(label1);
+		TextButton textButton1 = new TextButton("TextButton 1", skin);
+		table.add(textButton1);
+		
+		table.row();
+		
+		table.add("Custom components created from code").left().colspan(2);
+		table.row();
+		
+		// Create a FlashingLabelStyle from the default LabelStyle defined in uiskin.json
+		FlashingLabelStyle labelStyle = new FlashingLabelStyle(skin.get(LabelStyle.class));
+		
+		FlashingLabel label2 = new FlashingLabel("Label2", labelStyle);
+		table.add(label2);
+		TextButtonStyle textButtonStyle = skin.get("default", TextButtonStyle.class);
+		textButtonStyle.labelStyle = labelStyle;
+		TextButton textButton2 = new TextButton("TextButton 2", textButtonStyle);
+		table.add(textButton2);
+		
+		table.row();
+		
+		table.add("Custom styles can also be defined in the json.").left().colspan(2);
+	}
+
+	@Override
+	public void render () {
+		float delta = Gdx.graphics.getDeltaTime();
+		Gdx.gl.glClearColor(0f, 0f, 0f, 1f);
+		Gdx.gl.glClear(GL20.GL_COLOR_BUFFER_BIT);
+		
+		batch.begin();
+		stage.act(delta);
+		stage.draw();
+		batch.end();
+	}
+
+	@Override
+	public void dispose () {
+		batch.dispose();
+		stage.dispose();
+	}
+	
+	public static class FlashingLabel extends Label {
+		private float time = 0;
+		
+		public FlashingLabel (CharSequence text, Skin skin, String styleName) {
+			super(text, skin.get(styleName, FlashingLabelStyle.class));
+		}
+		
+		public FlashingLabel (CharSequence text, FlashingLabelStyle style) {
+			super(text, style);
+		}
+		
+		@Override
+		public void draw(Batch batch, float parentAlpha) {
+			FlashingLabelStyle style = (FlashingLabelStyle) getStyle();
+			time += Gdx.graphics.getDeltaTime();
+			if (time >= style.period) time -= style.period;
+			if (time <= style.period * style.duty) super.draw(batch, parentAlpha);
+		}
+		
+		static public class FlashingLabelStyle extends LabelStyle {
+			public float period = 1f;
+			public float duty = 0.5f;
```

Comment:
Regarding the fields, the benefit is that you can set them within the skin.json file.

But the real benefit of having a class which extends LabelStyle and overrides the createLabel method shows up when you create a TextButton using a TextButton style that has a value in the labelStyle field. With this method the TextButton can get an instance of FlashingLabel without knowing about its existence; we could also use the reflection to get an instance of a given Label subclass, but i think the first method is better (in my modest opinion).

Example:

    com.badlogic.gdx.tests.scenes.scene2d.ui.DynamicChildActorsTest$FlashingLabel$FlashingLabelStyle {
        flashing-label: {period: 2}
    }

For the TextButton:

    com.badlogic.gdx.scenes.scene2d.ui.TextButton.TextButtonStyle {
        flashing-button: {labelStyle: flashing-label}
    }

Then, in the code:

    TextButton textButton2 = new TextButton("TextButton 2", skin, "flashing-button");

I've written the above lines on the fly, if you want I can provide a working example.